src/app/Components/calendar-view/calendar-view.component.ts

```diff
@@ -36,11 +41,32 @@ export class CalendarViewComponent implements OnInit {
   }
 
   getWeekdayForFirstOfMonth(month: number) {
-    const day = new Date(this.year + '-' + month + '-01').getDay();
+    let day;
+
+    if (month <= this.month.id) {
+      day = new Date(this.year + '-' + month + '-01').getDay();
+    } else {
+      day = new Date(this.year + '-' + month + '-01').getDay() + 1;
+    }
 
     return day;
   }
 
+  getPreviousMonth() {
```

Comment:
this and line 59 arent get methods, rather they set the current month. something like `setToPreviousMonth` and `setToNextMonth` would be more appropriate